### PR TITLE
Support swagger extensions

### PR DESCRIPTION
**Pull request by tchssk**
This pull request extends `apidsl.Metadata()` to support [Swagger Extensions](https://github.com/OAI/OpenAPI-Specification/blob/master/guidelines/EXTENSIONS.md).
Now it includes only core implementations so I will add tests and comments for godoc.

https://gophers.slack.com/archives/goa/p1473747445000640

### CODE REVIEW

File: goagen/gen_swagger/swagger.go [hunk 1]
```diff
@@ -274,9 +286,70 @@ type (
 		Description string `json:"description,omitempty"`
 		// ExternalDocs is additional external documentation for this tag.
 		ExternalDocs *ExternalDocs `json:"externalDocs,omitempty"`
+		// Extensions defines the swagger extensions.
+		Extensions map[string]string `json:"-"`
 	}
+
+	_Info               Info
```
Comment (raphael) [inline goagen/gen_swagger/swagger.go]: Can you leave a comment explaining why the `_` types are needed?

File: goagen/gen_swagger/swagger.go [hunk 2]
```diff
@@ -274,9 +286,70 @@ type (
...
+		asserted[k] = v
+	}
+	merged, err := json.Marshal(asserted)
+	if err != nil {
+		return nil, err
+	}
+	return merged, nil
+}
+
+func (i Info) MarshalJSON() ([]byte, error) {
```
Comment (raphael) [inline goagen/gen_swagger/swagger.go]: Please add a comment for these

**Comment by tchssk** [comment]
Oops, I added the comments but travis tells me another error...

**Comment by raphael** [comment]
Pro tip: run `make` before pushing, that's what Travis does :)

**Comment by tchssk** [comment]
Thanks. I forget to `make` sometimes😓

**Comment by tchssk** [comment]
Thank you for your waiting. It is ready.

**Comment by raphael** [comment]
Do you think you could also send a PR against the v1 branch? this is a backwards compatible change and it would be great if people on the v1 branch could benefit from it too :)

**Comment by tchssk** [comment]
It's OK. But this PR is based on #775 so I will send that to v1 first.

**Comment by raphael** [comment]
Perfect, thanks again!

**Pull request merged**